test/teststl.cpp

```diff
@@ -3356,6 +3361,296 @@ class TestStl : public TestFixture {
               "}", true);
         ASSERT_EQUALS("", errout.str());
     }
+
+    void loopAlgoElementAssign() {
+        check("void foo() {\n"
+              "    for(int x:v)\n"
+              "        x = 1;\n"
+              "}\n",true);
+        ASSERT_EQUALS("[test.cpp:3]: (style) Consider using std::fill algorithm instead of a raw loop.\n", errout.str());
```

Comment:
how difficult would it be to printout the code that should be used in the message? If you think that makes the message too long... I am fine that we keep the short message like this and only printout the code when `--verbose` is used.

Reply:
I could print out a snippet of code, would that be on the second line? How would the message be, something like `It can be written as 'std::fill(std::begin(v), std::end(v), 1)'`?

The only issue is when it suggests two different algorithms. For example, there are some cases it suggests either `fill` or `generate` which depends if the expression has side effects. To avoid FPs, I suggest both. I could write two code snippets.

Reply:
If you write `reportError(.."abc\ndef"..);` then the short message will be `abc` and the verbose message will be `def`.

I think your suggestion sounds good. `It can be written as...`.

I also think that it sounds good to write both.